pkg/operator/api/api.go

```diff
@@ -125,12 +126,13 @@ func (c *Cluster) apiContainer(cluster *mon.ClusterInfo) v1.Container {
 	}
 }
 
-func (c *Cluster) startService(clientset *kubernetes.Clientset, clusterInfo *mon.ClusterInfo) error {
+func (c *Cluster) startService(clientset kubernetes.Interface, clusterInfo *mon.ClusterInfo) error {
 	labels := getLabels(clusterInfo.Name)
 	s := &v1.Service{
 		ObjectMeta: v1.ObjectMeta{
-			Name:   deploymentName,
-			Labels: labels,
+			Name:      deploymentName,
+			Namespace: c.Namespace,
```

Comment:
the mocked framework was more strict about setting the namespace on pods and services. I haven't seen a difference in behavior in the real world though.